src/stats/performance.test.ts

```diff
@@ -25,18 +26,56 @@ describe('calculate quarterly performance data of an episode', () => {
         const testdata = buildTestStructure([
             1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12,
         ])
-        const res = calcApplePodcastPerformanceQuarters({
-            performance: testdata,
-        })
+        const res = calcApplePodcastPerformanceQuarters(testdata)
         expect(res.maxListeners).toBe(12)
         expect(res.quarterMedianValues).toStrictEqual([2, 5, 8, 11])
     })
 
     it('only 3 values throws an exception', () => {
         const testdata = buildTestStructure([1, 2, 3])
-        // to catch an expection we have to wrap it in an anonymous fun
-        expect(() => {
-            calcApplePodcastPerformanceQuarters({ performance: testdata })
-        }).toThrow('too little data to calculate quarters')
+        // if the episode is too new we just set all values to 0
+        const res = calcApplePodcastPerformanceQuarters(testdata)
+        expect(res.maxListeners).toBe(0)
+        expect(res.quarterMedianValues).toStrictEqual([0, 0, 0, 0])
+    })
+
+    it('weird apple data having a longtail', () => {
+        const testdata = buildTestStructure([
+            1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 21, 1, 1, 1, 1, 1, 1, 1,
+        ])
+        const res = calcApplePodcastPerformanceQuarters(testdata)
+        expect(res.maxListeners).toBe(21)
+        expect(res.quarterMedianValues).toStrictEqual([2, 5, 8, 11])
+    })
+
+    it('weird apple data having a falling longtail', () => {
+        const testdata = buildTestStructure([
+            300, 250, 200, 190, 231, 210, 200, 200, 200, 50, 40, 30, 20, 16, 16,
+            16, 14, 10, 5, 1, 1, 1, 1, 1, 1,
+        ])
+        // expect a cut at 5% which is 231*0.05 = 11.55
+        const res = calcApplePodcastPerformanceQuarters(testdata)
+        expect(res.maxListeners).toBe(300)
+        expect(res.quarterMedianValues).toStrictEqual([225, 205, 45, 16])
+    })
+})
+
+describe('remove longtail from performance data', () => {
+    it('simple case', () => {
```

Comment:
While we're at it, let's add a few pathological cases. 😝 

```suggestion
   // Test with empty data
    it('empty data', () => {
        const testdata = []
        const res = removeLongtailFromPerformanceData(testdata, 0.05)
        expect(res.maxListeners).toBeUndefined()
        expect(res.performanceValues).toStrictEqual([])
    })

    // Test with only one item
    it('single item data', () => {
        const testdata = [1]
        const res = removeLongtailFromPerformanceData(testdata, 0.05)
        expect(res.maxListeners).toBe(1)
        expect(res.performanceValues).toStrictEqual([1])
    })

    // Test with negative numbers
    it('negative numbers', () => {
        const testdata = [1, 2, 3, -4, -5]
        const res = removeLongtailFromPerformanceData(testdata, 0.05)
        expect(res.maxListeners).toBe(3)
        expect(res.performanceValues.every((v) => v > 3 * 0.05)).toBe(true)
    })

    // Test with zeros
    it('zeroes in data', () => {
        const testdata = [0, 0, 0, 0, 0, 0]
        const res = removeLongtailFromPerformanceData(testdata, 0.05)
        expect(res.maxListeners).toBe(0)
        expect(res.performanceValues).toStrictEqual([0, 0, 0, 0, 0, 0])
    })

    // Test with all identical values
    it('all identical values', () => {
        const testdata = [5, 5, 5, 5, 5]
        const res = removeLongtailFromPerformanceData(testdata, 0.05)
        expect(res.maxListeners).toBe(5)
        expect(res.performanceValues).toStrictEqual([5, 5, 5, 5, 5])
    })
    
    it('simple case', () => {
```

Reply:
thanks, added some useful and defined tests and skipped the not defines/useless ones to dave some CO2 ;)